tinygrad/codegen/linearizer.py

```diff
@@ -411,17 +411,26 @@ def uop(self, uop:UOps, dtype:Optional[DType], vin:Tuple[UOp, ...], arg:Any=None
     if uop == UOps.CAST and all(x.uop == UOps.GEP for x in vin) and all_same([x.vin[0] for x in vin]) and all(x.arg == i for i,x in enumerate(vin)): return vin[0].vin[0]
     if uop == UOps.GEP and vin[0].uop == UOps.CONST: return self.const(vin[0].arg, dtype)
     if uop == UOps.ALU:
+      assert dtype is not None # for linter
       # rewrites. NOTE: the rewritten NEG op is still around...
       if arg == BinaryOps.ADD and vin[1].uop == UOps.ALU and vin[1].arg == UnaryOps.NEG: return self.uop(UOps.ALU, dtype, (vin[0], vin[1].vin[0]), BinaryOps.SUB, cachable=cachable)
       # constant folding
       if arg == UnaryOps.NEG and vin[0].uop == UOps.CONST: return self.const(-vin[0].arg, dtype)
+      if arg in BinaryOps and vin[0].uop == UOps.CONST and vin[1].uop == UOps.CONST:
+        fs = {BinaryOps.ADD: lambda a, b: a + b, BinaryOps.SUB: lambda a, b: a - b, BinaryOps.MUL:lambda a, b: a * b, BinaryOps.DIV:lambda a, b: a / b if b != 0. else float('nan'), BinaryOps.MAX: lambda a,b: max(a,b), BinaryOps.CMPLT: lambda a,b: int(a < b), BinaryOps.MOD: lambda a,b: a % b} #pylint: disable=unnecessary-lambda
```

Comment:
This is not correct in all cases and needs *much* more comprehensive testing. Example of issue: not all dtypes are float32

Reply:
Ah I see. Will add much more tests comparing it with numpy outputs!

Reply:
There are going to quite some exceptions to get the same behaviour as numpy. Do we want that? Or I could convert a,b to numpy here to easily fix it

Reply:
I also have to create another env parameter such that constant folding can be skipped to verify all the backends easily in test_ops.py.
It feels this gets too complex for the small benefit of some optimized constants.